R/grid_performance.R

```diff
@@ -125,3 +138,69 @@ estimate_class_prob <- function(dat, metric, param_names, outcome_name,
       event_level = event_level
     )
 }
+
+# ------------------------------------------------------------------------------
+
+estimate_surv <- function(dat, metric, param_names, outcome_name, case_weights, types) {
+  #  potentially need to work around submodel parameters since those are within .pred
+  dat <- unnest_parameters(dat, param_names)
+  dat %>%
+    dplyr::group_by(!!!rlang::syms(param_names)) %>%
+    metric(
+      truth = !!rlang::sym(outcome_name),
+      estimate = !!maybe_estimate(metric),
+      case_weights = !!case_weights,
+      !!maybe_surv_prob(metric)
+    )
+}
+
+unnest_parameters <- function(x, params = NULL) {
+  # When muti_predict is used, .pred will have the tuning parameter values.
+  # Other (non-submodel) parameters will be outside of 'x'.
+  # If this happens, pull the submodel parameters out of .pred and put them at
+  # the out level ('x') with the rest (if any)
+  outer_nms <- names(x)
+  inner_nms <- names(x$.pred[[1]])
+  has_inner_params <- any(params %in% inner_nms)
+  has_pred <- any(outer_nms == ".pred")
+  if (is.null(params) || !has_pred || !has_inner_params) {
+    return(x)
+  }
```

Comment:
```suggestion
   if (is.null(params)) {
    return(x)
   }
   
  # When multi_predict() is used, .pred will have the tuning parameter values.
  # Other (non-submodel) parameters will be outside of 'x'.
  # If this happens, pull the submodel parameters out of .pred and put them at
  # the out level ('x') with the rest (if any)
  
  outer_nms <- names(x)
  has_pred <- any(outer_nms == ".pred")
   if (!has_pred) {
    return(x)
  }
  
  inner_nms <- names(x$.pred[[1]])
  has_inner_params <- any(params %in% inner_nms)
  if (!has_inner_params) {
    return(x)
  }

```

Reply:
@topepo this is where the warning `Unknown or uninitialised column: `.pred`.` came from. With this suggestion, the early returns are as early as they can be, hence not looking for `names(x$.pred[[1]])` when there is no `.pred` column.